### PR TITLE
licenses: Update BSL change date for master/24.1

### PR DESCRIPTION
**Pull request by jayshrivastava**
This commit updates the BSL license for master/24.1.

Release note: None

### CODE REVIEW

File: licenses/BSL.txt [hunk 1]
```diff
@@ -15,7 +15,7 @@ Additional Use Grant: You may make use of the Licensed Work, provided that
                       Licensed Work by creating tables whose schemas are
                       controlled by such third parties.
 
-Change Date:          2026-10-01
+Change Date:          2027-04-01
```
Comment (celiala) [inline licenses/BSL.txt:18]: hi @jayshrivastava !

can you also advance the version for "Licensed Work" from 23.2 to 24.1? (line 6)

(could you also update the runbook for this task?)

everything else LGTM!

thank you!

https://github.com/cockroachdb/cockroach/commit/5d1f4bbeb167484d5ae7f3257c15880fb6f4cbc2

Comment (jayshrivastava) [reply licenses/BSL.txt:18]: Done!

**Review by celiala — APPROVED** [review]
thank you -- LGTM! :shipit:

**Pull request merged**